nvd/matcher.go

```diff
@@ -0,0 +1,30 @@
+package nvd
+
+import "strings"
+
+type Matcher interface {
+	Matches(CVEEntry) bool
+}
+
+var _ Matcher = (*cpeMatcher)(nil)
+
+type cpeMatcher struct {
+	substring string
+}
+
+func NewCPEMatcher(substring string) cpeMatcher {
```

Comment:
Don't return an unexported type in an exported function -- just mark the return type as `Matcher`. (And if you do, you can remove the compile-time assertion above.)